exec/kubernetes_deploy.go

```diff
@@ -2,35 +2,90 @@ package exec
 
 import (
 	"bytes"
+	"errors"
 	"fmt"
-	"github.com/portainer/agent"
 	"os"
 	"os/exec"
 	"path"
 	"runtime"
 	"strings"
+
+	"github.com/portainer/agent"
 )
 
 // KubernetesDeployer represents a service to deploy resources inside a Kubernetes environment.
 type KubernetesDeployer struct {
-	binaryPath string
+	command string
 }
 
 // NewKubernetesDeployer initializes a new KubernetesDeployer service.
 func NewKubernetesDeployer(binaryPath string) *KubernetesDeployer {
+	command := path.Join(binaryPath, "kubectl")
+	if runtime.GOOS == "windows" {
+		command = path.Join(binaryPath, "kubectl.exe")
+	}
+
 	return &KubernetesDeployer{
-		binaryPath: binaryPath,
+		command: command,
 	}
 }
 
 // Deploy will deploy a Kubernetes manifest inside a specific namespace
 // it will use kubectl to deploy the manifest.
-func (deployer *KubernetesDeployer) Deploy(token, data string, namespace string) ([]byte, error) {
-	command := path.Join(deployer.binaryPath, "kubectl")
-	if runtime.GOOS == "windows" {
-		command = path.Join(deployer.binaryPath, "kubectl.exe")
+// kubectl uses in-cluster config.
+func (deployer *KubernetesDeployer) Deploy(name string, filePaths []string, prune bool) error {
+	if len(filePaths) == 0 {
+		return errors.New("missing file paths")
+	}
+
+	stackFilePath := filePaths[0]
+
+	args := make([]string, 0)
+	// Specifying "--insecure-skip-tls-verify" make kubectl return error "default cluster has no server defined"
+	//args = append(args, "--insecure-skip-tls-verify")
+	args = append(args, "--namespace", "default")
+	args = append(args, "apply", "-f", stackFilePath)
+
+	var stderr bytes.Buffer
+	cmd := exec.Command(deployer.command, args...)
+	cmd.Stderr = &stderr
+
+	_, err := cmd.Output()
+	if err != nil {
+		return fmt.Errorf("failed deploying kubernetes stack %w: %s", err, stderr.String())
 	}
 
+	return nil
+}
+
+func (deployer *KubernetesDeployer) Remove(name string, filePaths []string) error {
+	if len(filePaths) == 0 {
+		return errors.New("missing file paths")
+	}
+
+	stackFilePath := filePaths[0]
+
+	args := []string{}
+
+	args = append(args, "--namespace", "default")
+	args = append(args, "delete", "-f", stackFilePath)
+
+	var stderr bytes.Buffer
+	cmd := exec.Command(deployer.command, args...)
+	cmd.Stderr = &stderr
+
+	_, err := cmd.Output()
+	if err != nil {
+		return fmt.Errorf("failed removing kubernetes stack %w: %s", err, stderr.String())
+	}
+
+	return nil
+}
+
+// DeployRawConfig will deploy a Kubernetes manifest inside a specific namespace
+// it will use kubectl to deploy the manifest and receives a raw config.
+// kubectl uses in-cluster config.
+func (deployer *KubernetesDeployer) DeployRawConfig(token, config string, namespace string) ([]byte, error) {
 	args := make([]string, 0)
```

Comment:
DeployRawConfig and Deploy share many common codes. It would be nice if pull them into a sharing method.